File: BackEnd/src/features/record/schema.py
from uuid import UUID

from pydantic import BaseModel


class RecordCreate(BaseModel):
    name: str
    service_id: str | None = None
    status: bool
    optional_status: str | None = None
    type_tag: str
    value: str
    notes: dict[str, str | int | float] | None = None
    entities: list[UUID]


class RecordUpdate(BaseModel):
    id: UUID
    name: str | None = None
    service_id: str | None = None
    status: bool | None = None
    optional_status: str | None = None
    type_tag: str | None = None
    value: str | None = None
    notes: dict[str, str | int | float] | None = None

File: SPA/src/features/records/schema.ts
import z from "zod/v4";

export const RecordSchema = z.object({
  id: z.uuid(),
  type_tag: z.string(),
  name: z.string(),
  status: z.boolean(),
  optional_status: z.string().optional(),
  service_id: z.string().optional(),
});

export type RecordType = z.infer<typeof RecordSchema>;
